app/src/main/java/com/educacionit/biciya/home/presenter/MapPresenterImpl.kt

```diff
@@ -0,0 +1,31 @@
+package com.educacionit.biciya.home.presenter
+
+import com.educacionit.biciya.home.contracts.map.model.MapModel
+import com.educacionit.biciya.home.contracts.map.MapPresenter
+import com.educacionit.biciya.home.contracts.map.MapView
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+
+class MapPresenterImpl(private val view : MapView,
+                       private val model : MapModel
+) : MapPresenter{
+    override fun loadStations() {
+        view.setLoadingVisibility(true)
+        CoroutineScope(Dispatchers.IO).launch {
+            try {
+                val stations = model.getStations()
+                CoroutineScope(Dispatchers.Main).launch {
+                    view.showStationOnMap(stations!!.data.stations)
+                    view.setLoadingVisibility(false)
+                }
+            } catch (e: Exception){
+                CoroutineScope(Dispatchers.Main).launch {
+                    view.showErrorMessage(e.message.toString())
+                    view.setLoadingVisibility(false)
+                }
+            }
+        }
```

Comment:
We can mark this function as suspend and only launch a new coroutine for the model.getStations() call and avoid to create another main coroutine.